### PR TITLE
Add zoom handler with example

### PR DESCRIPTION
**Pull request by HavardNJ**

### CODE REVIEW

File: .storybook/src/grid-layer.stories.ts [hunk 1]
```diff
@@ -47,8 +47,8 @@ const createEventObj = (elm: any) => {
     .range([0, 500]);
 
   return {
-    xscale: xscale.copy(),
-    yscale: yscale.copy(),
+    xScale: xscale.copy(),
```
Comment (stalenestas) [inline .storybook/src/grid-layer.stories.ts]: Should the other xscale be changed too?

File: .storybook/src/zoom.stories.ts [hunk 2]
```diff
@@ -0,0 +1,285 @@
+import * as d3 from 'd3';
```
Comment (stalenestas) [inline .storybook/src/zoom.stories.ts]: Try to import only the used method instead of everything in d3

File: .storybook/src/zoom.stories.ts [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+  render(event: OnRescaleEvent | OnUpdateEvent) {
+    const {
+      ctx,
+    } = this;
+
+    if (!ctx) return;
+
+    // Get the bounds area
+    const xb1 = event.xBounds[0];
```
Comment (stalenestas) [inline .storybook/src/zoom.stories.ts]: Could be destructed const [xb1, xb2] = event.xBounds

File: .storybook/src/zoom.stories.ts [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    this.render(event);
+  }
+
+  onUpdate(event: OnUpdateEvent) {
+    super.onUpdate(event);
+
+    this.render(event);
+  }
+
+  render(event: OnRescaleEvent | OnUpdateEvent) {
```
Comment (stalenestas) [inline .storybook/src/zoom.stories.ts]: Should be refactored into smaller methods

File: src/control/ZoomPanHandler.ts [hunk 5]
```diff
@@ -0,0 +1,480 @@
...
+      .on('zoom', this.onZoom);
+
+      this.container.call(this.zoom);
+  }
+
+  /**
+   * Handle zoom
+   */
+  onZoom(): void{
+    if(!d3.event)return;
```
Comment (stalenestas) [inline src/control/ZoomPanHandler.ts]: Always brackets { return }

Comment (HavardNJ) [reply src/control/ZoomPanHandler.ts]: I think we need to establish which linting rules to follow.

File: .storybook/src/grid-layer.stories.ts (hunk 1 above)
Comment (HavardNJ) [reply .storybook/src/grid-layer.stories.ts]: Given that this is a storybook example and the the variable is a copy, I don't think it's necessary. But we should probably discuss naming conventions of variables and functions to be consistent.

File: .storybook/src/zoom.stories.ts (hunk 3 above)
Comment (HavardNJ) [reply .storybook/src/zoom.stories.ts]: Fixed

File: .storybook/src/zoom.stories.ts (hunk 2 above)
Comment (HavardNJ) [reply .storybook/src/zoom.stories.ts]: Removed import as it was not in use. We need to set up linting.

File: .storybook/src/zoom.stories.ts (hunk 4 above)
Comment (HavardNJ) [reply .storybook/src/zoom.stories.ts]: I cleaned this up and shortened it a bit, but given that this is a (slightly silly) example which is not intended to be reused or in need of much maintenance, I don't see the need. The whole TestLayer should not be in this file if we were to adder to best practices, but for a storybook example I think it makes it easier to have all the code in one place and to read the drawing code sequentially.

File: .storybook/src/zoom.stories.ts (hunk 2 above)
Comment (HavardNJ) [reply .storybook/src/zoom.stories.ts]: Rebased and fixed d3 imports for ZoomPanHandler

**Pull request merged**